python/pyarrow/parquet.py

```diff
@@ -1958,8 +1958,11 @@ def write_to_dataset(table, root_path, partition_cols=None,
             "implementation."
         )
         metadata_collector = kwargs.pop('metadata_collector', None)
+        file_visitor = None
         if metadata_collector is not None:
-            raise ValueError(msg.format("metadata_collector"))
+            def file_visitor(written_file):
+                if written_file.metadata:
```

Comment:
Is this `if` check needed (in theory)? Since this will always be parquet files in this case

Reply:
I suppose not, I have removed it.

Reply:
It could instead be an `assert`